src/Mailer/Transport/MailgunTransport.php

```diff
@@ -156,6 +156,28 @@ public function send(Email $email)
         return $res;
     }
 
+    /**
+     * Add file attachment to email.
+     *
+     * @param string $partName Name of the file part
+     * @param array $attachment Attachment as initially set via setAttachment()
+     * @param string $fileName Desired filename of the attachment
+     * @return \Cake\Http\Client\FormDataPart
+     */
+    protected function _addFile($partName, $attachment, $fileName = '')
+    {
+        if (isset($attachment['file'])) {
+            $file = $this->_formData->addFile($partName, fopen($attachment['file'], 'r'));
+        } else {
```

Comment:
How would you make sure that the `data` attribute is present?

Reply:
It looks like Cake's `Email` class already takes care of that here: https://github.com/cakephp/cakephp/blob/master/src/Mailer/Email.php#L1818-L1821 .  However, I can add an additional check if that's preferred?  Let me know :-)

Reply:
That's fine!